include/pokemon.h

```diff
@@ -337,17 +337,27 @@ struct SpeciesInfo /*0x24*/
 struct BattleMove
 {
     u16 effect;
-    u16 power;  //higher than 255 for z moves
-    u8 type;
+    u8 power;  //higher than 255 for z moves
+    u8 type:5;
+    u8 recoil:3;
+
     u8 accuracy;
-    u8 pp;
     u8 secondaryEffectChance;
-    u16 target;
     s8 priority;
+    u8 multihit:4;
+    u8 critBoost:2;
+    u8 split:2;
+
+    u16 target;
+    u8 pp;
+    union {
+        u8 effect;
+        u8 powerOverride;
+    } zMove;
+
     u32 flags;
-    u8 split;
+
     u16 argument;
-    u8 zMoveEffect;
 };
```

Comment:
Preliminary thoughts:
- `recoil` doesn't have enough bits to fit all the values possible by `RECOIL_PERCENTAGE`, like 1%, so there's no point of having macro anyway if you're going to need enough bits to store at least a value of 100.
  - On that same note, I don't see the need the point of using bitfields for `type`, `split` and `critBoost`. I rather have the fields be in the same order as they are in `battle_moves.h`, since you're barely saving space by changing those.
- Fields should use either defines or numbers, but not both like how `multihit` is doing.
- Good call on reducing `power` back to u8, though you forgot to remove the comment.
- `

Reply:
- I envisioned `recoil` as actually only containing the _fraction_ of damage taken rather than a % so only having values from 1-7 would be fine. I put in the RECOIL_PERCENTAGE macro just to make it crystal clear, but you're right in that it implies any % from 1-100 works. I'll just rename the field to `recoilFraction`.
- Bitfields aren't necessary, to me they're just neat. I'll give them each a proper u8.
- The `multihit` field is special because a value of 1, which would otherwise be totally meaningless, is repurposed to signify a multihit move that hits a random number of times. The define is there, again, to make it clear that that's what it's for.
- Oops
- `